src/geo/data-providers/geojson/geojson-data-provider.js

```diff
@@ -15,23 +15,34 @@ var GeoJSONDataProvider = function (vectorLayerView, layerIndex) {
 // TODO: We can extract each "generator" to an individual file so that this file doesn't get too BIG
 GeoJSONDataProvider.prototype._dataGeneratorsForDataviews = {
   category: function (features, options) {
```

Comment:
There're a couple of things I don't like about this function:

1. It acts differently depending on the `filterEnabled` option: If this option is true, it uses the `features` to generate the data that the dataview needs; but the option is false, it uses the filter to get all the data (no filters applied) and generate the data for the dataview (two very different responsibilities).
2. We're directly passing the dataview attributes through, which are subject to change....
3. This file is getting too big and we're not testing these methods, which are starting to get complicated.

Will try to solve these issues in a separate PR.

cc: @fdansv @javisantana